youtube_dl/extractor/vxxx.py

```diff
@@ -0,0 +1,240 @@
+
+    _BASE_URL = 'https://vxxx.com'
+    _INFO_OBJECT_URL_TMPL = '{0}/api/json/video/86400/0/{1}/{2}.json'
+    _FORMAT_OBJECT_URL_TMPL = '{0}/api/videofile.php?video_id={1}'
+
+    def _download_info_object(self, video_id):
+        return self._download_json(
+            self._INFO_OBJECT_URL_TMPL.format(
+                self._BASE_URL,
+                int(video_id) // 1000 * 1000,
+                video_id,
+            ), video_id, headers={'Referer': self._BASE_URL})['video']
+
+    def _download_format_object(self, video_id):
+        return self._download_json(
+            self._FORMAT_OBJECT_URL_TMPL.format(self._BASE_URL, video_id),
+            video_id,
+            headers={'Referer': self._BASE_URL}
+        )
+
+    def _decode_base164(self, e):
+        """
+        Some non-standard encoding called "base164" in the JavaScript code. It's
+        similar to the regular base64 with a slightly different alphabet:
+            - "АВСЕМ" are Cyrillic letters instead of uppercase Latin letters
+            - "." is used instead of "+"; "," is used instead of "/"
+            - "~" is used for padding instead of "="
+        """
+
+        # using the kwarg to memoise the result
+        def get_trans_tbl(from_, to, tbl={}):
+            k = (from_, to)
+            if not tbl.get(k):
+                tbl[k] = str.maketrans(from_, to)
+            return tbl[k]
+
+        trans_tbl = get_trans_tbl(
+            '\u0410\u0412\u0421\u0415\u041c.,~',
+            'ABCEM+/=')
+        return base64.b64decode(e.translate(trans_tbl)).decode()
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        info_object = self._download_info_object(video_id)
+
+        title = info_object['title']
+        stats = info_object.get('statistics') or {}
+        info = {
+            'id': video_id,
+            'title': title,
+            'display_id': info_object.get('dir'),
+            'thumbnail': url_or_none(info_object.get('thumb')),
+            'description': strip_or_none(info_object.get('description')) or None,
+            'timestamp': unified_timestamp(info_object.get('post_date')),
+            'duration': parse_duration(info_object.get('duration')),
+            'view_count': int_or_none(stats.get('viewed')),
+            'like_count': int_or_none(stats.get('likes')),
+            'dislike_count': int_or_none(stats.get('dislikes')),
+            'average_rating': float_or_none(stats.get('rating')),
+            'categories': [category['title'] for category in (info_object.get('categories') or {}).values()
+                           if category.get('title')],
+            'age_limit': 18,
+        }
+
+        format_object = self._download_format_object(video_id)
+        m3u8_formats = self._extract_m3u8_formats(
+            '{0}/{1}&f=video.m3u8'.format(
+                self._BASE_URL,
+                self._decode_base164(format_object[0]['video_url'])
+            ),
+            video_id, 'mp4')
```

Comment:
Let's try this.
```suggestion
            video_id, 'mp4', entry_protocol='m3u8_native')
```
Otherwise the download tests will have to be tweaked to skip the actual download.